src/lib.rs

```diff
@@ -1313,6 +1313,14 @@ impl Build {
                 self.cargo_output
                     .print_metadata(&format_args!("cargo:rustc-link-lib={}", stdlib));
             }
+            // Link c++ lib from WASI sysroot
+            if self.get_target()?.contains("wasi") {
+                let wasi_sysroot = self.wasi_sysroot()?;
+                self.cargo_output.print_metadata(&format_args!(
+                    "cargo:rustc-flags=-L {}/lib/wasm32-wasi -lstatic=c++ -lstatic=c++abi",
```

Comment:
Took another look and realise it is hardcoded to wasi.

I don't like it, but I think it's ok given that the preview wasi is also wasi.